modules/charon-core/src/main/java/org/wso2/charon3/core/config/SCIMCustomSchemaExtensionBuilder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.charon3.core.config;
+
+import org.wso2.charon3.core.attributes.SCIMCustomAttribute;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.schema.AttributeSchema;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This builds the custom user schema.
+ */
+public class SCIMCustomSchemaExtensionBuilder extends ExtensionBuilder {
+
+    private Map<String, ExtensionAttributeSchemaConfig> customConfig;
+    private static SCIMCustomSchemaExtensionBuilder customSchemaExtensionBuilder =
+            new SCIMCustomSchemaExtensionBuilder();
+    private Map<String, AttributeSchema> attributeSchemaConfig;
```

Comment:
Any reason to define this in class level? Can move to the method level

Reply:
fixed with https://github.com/wso2/charon/pull/323/commits/c1599064a88bc6e58faf5a8017933e1f1cc86d3c